images/virtualization-artifact/pkg/controller/service/disk_service.go

```diff
@@ -251,6 +256,85 @@ func (s DiskService) GetCapacity(pvc *corev1.PersistentVolumeClaim) string {
 	return ""
 }
 
+func (s DiskService) GetStorageProfile(ctx context.Context, storageClassName *string) (*cdiv1.StorageProfile, error) {
+	sc, err := s.GetStorageClass(ctx, storageClassName)
+	if err != nil {
+		return nil, err
+	}
+	var sp cdiv1.StorageProfile
+	if err = s.client.Get(ctx, types.NamespacedName{Name: sc.GetName()}, &sp); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, fmt.Errorf("storageProfile not found: %w", err)
+		}
+		return nil, err
+	}
+	return &sp, nil
+}
+
+type StorageCapabilities struct {
+	AccessMode corev1.PersistentVolumeAccessMode
+	VolumeMode corev1.PersistentVolumeMode
+}
+
+func (cp StorageCapabilities) IsEmpty() bool {
+	return cp.AccessMode == "" && cp.VolumeMode == ""
+}
+
+func (s DiskService) parseVolumeMode(status cdiv1.StorageProfileStatus) StorageCapabilities {
+	accessModeWeights := map[corev1.PersistentVolumeAccessMode]int{
+		corev1.ReadOnlyMany:     0,
+		corev1.ReadWriteOncePod: 1,
+		corev1.ReadWriteOnce:    2,
+		corev1.ReadWriteMany:    3,
+	}
+
+	volumeModeWeights := map[corev1.PersistentVolumeMode]int{
+		corev1.PersistentVolumeFilesystem: 0,
+		corev1.PersistentVolumeBlock:      1,
+	}
+
+	getAccessModeMax := func(modes []corev1.PersistentVolumeAccessMode) corev1.PersistentVolumeAccessMode {
+		weight := -1
+		var m corev1.PersistentVolumeAccessMode
+		for _, mode := range modes {
+			if accessModeWeights[mode] > weight {
+				weight = accessModeWeights[mode]
+				m = mode
+			}
+		}
+		return m
+	}
+
+	var scaps []StorageCapabilities
```

Comment:
scaps? The naming is not obvious, better storageCapabilities

Reply:
scaps: s - storage, caps - capabilities.
So, for me it is obvious, but I will rename it

Reply:
fixed